packages/client/test/net/protocol/snapprotocol.spec.ts

```diff
@@ -456,6 +456,86 @@ tape('[SnapProtocol]', (t) => {
     )
     t.end()
   })
+
+  t.test('GetTrieNodes should encode/decode correctly', (t) => {
+    const config = new Config({ transports: [] })
+    const chain = new Chain({ config })
+    const p = new SnapProtocol({ config, chain })
+
+    const reqId = BigInt(1)
+    const root = Buffer.from(
+      '04157502e6177a76ca4dbf7784e5ec1a926049db6a91e13efb70a095a72a45d9',
+      'hex'
+    )
+    const paths = [[Buffer.from('0x0', 'hex')], [Buffer.from('0x0', 'hex')]]
+    const bytes = BigInt(5000000)
+
+    const payload = p.encode(p.messages.filter((message) => message.name === 'GetTrieNodes')[0], {
```

Comment:
I'm not entire sure what this line does (looks a bit like magic to me), it filters out the right message name and then assigns the values to it..? How does this magically call the `encode` then..?